src/biogeochem/CNPhenologyMod.F90

```diff
@@ -972,6 +1034,147 @@ subroutine CNSeasonDecidPhenology (num_soilp, filter_soilp       , &
+    logical :: do_onset                       ! Flag if onset should happen (return value)
+    !
+    ! !LOCAL VARIABLES:
+    real(r8), parameter :: snow5d_thresh_for_onset        = 0.1_r8          ! 5-day snow depth threshold for onset
+    real(r8), parameter :: min_critial_daylength_onset    = 39300._r8/2._r8 !  Minimum daylength for onset to happen
+    !-----------------------------------------------------------------------
+
+    do_onset = .false.
+    ! Test to turn on growing degree-day sum, if off.
+    ! switch on the growing degree day sum on the winter solstice
+
+    if (onset_gddflag == 0._r8 .and. ws_flag == 1._r8) then
+        onset_gddflag = 1._r8
+        onset_gdd = 0._r8
+    end if
+
+    ! Test to turn off growing degree-day sum, if on.
+    ! This test resets the growing degree day sum if it gets past
+    ! the summer solstice without reaching the threshold value.
+    ! In that case, it will take until the next winter solstice
+    ! before the growing degree-day summation starts again.
+
+    if (onset_gddflag == 1._r8 .and. ws_flag == 0._r8) then
+        onset_gddflag = 0._r8
+        onset_gdd = 0._r8
+    end if
+
+    ! if the gdd flag is set, and if the soil is above freezing
+    ! then accumulate growing degree days for onset trigger
+
+    if (onset_gddflag == 1.0_r8 .and. soilt > SHR_CONST_TKFRZ) then
+        onset_gdd = onset_gdd + (soilt-SHR_CONST_TKFRZ)*fracday
+    end if
+    if ( onset_thresh_depends_on_veg) then
+       ! separate into non-arctic seasonally deciduous pfts
+       ! (temperate broadleaf deciduous
+       ! tree) and arctic/boreal seasonally deciduous pfts (boreal
+       ! needleleaf deciduous tree,
+       ! boreal broadleaf deciduous tree, boreal broadleaf deciduous
+       ! shrub, C3 arctic grass)
+       if (onset_gdd > crit_onset_gdd .and.  season_decid_temperate == 1) then
+           do_onset = .true.
+        ! Note: The min_critial_daylength_onset is because for some coastal
+        ! points the other triggers could allow onset in January/February
+        ! which isn't sustainable and is a degenerate case. To prevent this
+        ! condition was added, but now the other conditions aren't triggered
+        ! until much later so it's value just needs to be high enough to prevent
+        ! the degenerate case of happening too early, and low enough that it
+        ! doesn't restrict onset. As such the value of this parameter shouldn't
+        ! matter for reasonable values between the two degnerate cases.
+        else if (season_decid_temperate == 0 .and.  onset_gddflag == 1.0_r8 .and. &
+                soila10 > SHR_CONST_TKFRZ .and. &
+                t_a5min > SHR_CONST_TKFRZ .and. ws_flag==1.0_r8 .and. &
+                dayl>min_critial_daylength_onset .and.  snow_5day<snow5d_thresh_for_onset) then
+           do_onset = .true.
+        end if
+    else
+       ! set onset_flag if critical growing degree-day sum is exceeded
```

Comment:
We are setting do_onset here, not onset_flag

Reply:
Got it.